pkg/apis/core/validation/validation.go

```diff
@@ -421,6 +421,33 @@ func ValidateVolumes(volumes []core.Volume, podMeta *metav1.ObjectMeta, fldPath
 	return vols, allErrs
 }
 
+func ValidateVolumesUpdate(volumes, oldVolumes []core.Volume, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	mapOldVolumeNames := make(map[string]core.Volume)
+	for _, oldVol := range oldVolumes {
+		o := oldVol.DeepCopy()
+		if o.VolumeSource.Ephemeral == nil {
+			continue
+		}
+		// If the old volume has limits set, then we will skip the validation for the volume, but verify that
+		// for any old volume that does not have a limit.
+		if _, ok := o.VolumeSource.Ephemeral.VolumeClaimTemplate.Spec.Resources.Limits[core.ResourceStorage]; ok {
+			continue
+		}
+		mapOldVolumeNames[o.Name] = *o
+	}
+	for i, vol := range volumes {
+		oldVol, ok := mapOldVolumeNames[vol.Name]
+		if !ok {
```

Comment:
If there is an update that creates a volume in the spec, will that get checked for not having limits? I don't see where that's done.

See comment below in the tests.

Reply:
Yeah, the check is not here, but in the previous `validatePodMetadataAndSpec` , `ValidateDaemonSetSpec`, `ValidateDeploymentSpec`, `ValidateXXXSpec`, etc